frontend/src/metabase/dashboard/components/DashboardGrid.jsx

```diff
@@ -70,31 +69,47 @@ export default class DashboardGrid extends Component {
   UNSAFE_componentWillReceiveProps(nextProps) {
     this.setState({
       dashcards: this.getSortedDashcards(nextProps),
-      layout: this.getLayout(nextProps),
+      layouts: this.getLayouts(nextProps),
     });
   }
 
-  onLayoutChange(layout) {
-    const changes = layout.filter(
-      newLayout =>
-        !_.isEqual(newLayout, this.getLayoutForDashCard(newLayout.dashcard)),
-    );
-    for (const change of changes) {
-      this.props.setDashCardAttributes({
-        id: change.dashcard.id,
-        attributes: {
-          col: change.x,
-          row: change.y,
-          sizeX: change.w,
-          sizeY: change.h,
-        },
-      });
+  onLayoutChange = ({ layout, breakpoint }) => {
+    // We allow moving and resizing cards only on the desktop
+    // Ensures onLayoutChange triggered by window resize,
+    // won't break the main layout
+    if (breakpoint !== "desktop") {
+      return;
     }
 
-    if (changes && changes.length > 0) {
+    const { dashboard, setMultipleDashCardAttributes } = this.props;
+    const changes = [];
+
+    layout.forEach(layoutItem => {
+      const dashboardCard = dashboard.ordered_cards.find(
+        card => String(card.id) === layoutItem.i,
+      );
+      const changed = !_.isEqual(
```

Comment:
Nit-picking: wdyt of a linespace between these const declarations, the if below etc?

Reply:
I'm not very opinionated here, can add 🙂